src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -0,0 +1,442 @@
+        Condition="Exists($(PublishDir))"
+        Directories="$(PublishDir)" />
+
+    <RemoveDir
+        Condition="Exists($(_CrossProjFileDir))"
+        Directories="$(_CrossProjFileDir)" />
+
+    <MakeDir Directories="$(PublishDir)" />
+
+  </Target>
+  
+  <!--
+    ============================================================
+                                        RestoreForComposeCache
+
+    Preps the Restore Task.
+    ============================================================
+    -->
+  <Target Name="RestoreForComposeCache">
+
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+                 Targets="Restore"
+                 Properties="RestoreGraphProjectInput=$(MSBuildProjectFullPath);
+                             DisableImplicitFrameworkReferences=true;
+                             TargetFramework=$(_TFM);"/>
+
+  </Target>
+
+
+  <!--
+    ============================================================
+                                        ComputeAndCopyFilesToCacheDirectory
+
+    Computes the list of all files to copy to the publish directory and then publishes them.
+    ============================================================
+    -->
+  <Target Name="ComputeAndCopyFilesToCacheDirectory"
+          DependsOnTargets="ComputeFilesToCache;
+                            CopyFilesToCacheDirectory" />
+
+  <!--
+    ============================================================
+                                        CopyFilesToCacheDirectory
+
+    Copy all build outputs, satellites and other necessary files to the publish directory.
+    ============================================================
+    -->
+  <Target Name="CopyFilesToCacheDirectory"
+          DependsOnTargets="_CopyResolvedManagedFiles;
+                            _CopyResolvedUnOptimizedFiles">
+
+    <Message Text="Files were composed in $(PublishDir)"
+             Importance="high"/>
+    <RemoveDir
+        Condition="'$(PreserveCrossProjFolder)' == ''"
+        Directories="$(_CrossProjFileDir)" />
+  </Target>
+
+  <!--
+    ============================================================
+                                        _CopyResolvedManagedFiles
+
+    Copy _ManagedResolvedFileToPublish items to the publish directory.
+    ============================================================
+    -->
+  <Target Name="_CopyResolvedManagedFiles"
+          DependsOnTargets="_ComputeResolvedFilesToCacheTypes;
+                            _RunOptimizer"
+          Inputs="@(_ManagedResolvedFileToPublish)"
+          Outputs="@(_ManagedResolvedFileToPublish->'$(PublishDir)%(RecursiveDir)%(Fileame)%(Extension)')">
+
+    <!--
+        Not using SkipUnchangedFiles="true" because the application may want to change
+        one of these files and not have an incremental build replace it.
+        -->
+    <Copy SourceFiles = "@(_ManagedResolvedFileToPublish)"
+          DestinationFiles="$(PublishDir)%(RecursiveDir)%(Filename)%(Extension)"
```

Comment:
We fixed a bug the way Publish copies files and the bug seems to be duplicated here.  See [https://github.com/dotnet/sdk/pull/581](https://github.com/dotnet/sdk/pull/581)